accesscontroltool-startuphook-bundle/src/main/java/biz/netcentric/cq/tools/actool/startuphook/impl/AcToolStartupHookServiceImpl.java

```diff
@@ -177,4 +146,28 @@ private void copyAcHistoryToOrFromApps(boolean isCloudReady) {
 
     }
 
+    @Override
+    public void processRepository(SlingRepository repo) throws Exception {
+        boolean isCloudReady = RuntimeHelper.isCloudReadyInstance();
+        LOG.info("AcTool Startup Hook (isCloudReady: {}  activationMode: {})", 
+                isCloudReady,
+                activationMode);
+
+        boolean applyOnStartup = (activationMode == Config.StartupHookActivation.ALWAYS) 
+                || (isCloudReady && activationMode == Config.StartupHookActivation.CLOUD_ONLY);
+
+        if(applyOnStartup) {
+            List<String> relevantPathsForInstallation = getRelevantPathsForInstallation(repo);
+            LOG.info("Running AcTool with "
+                    + (relevantPathsForInstallation.isEmpty() ? "all paths" : "paths " + relevantPathsForInstallation) + "...");
+            acInstallationService.apply(null, relevantPathsForInstallation.toArray(new String[relevantPathsForInstallation.size()]),
```

Comment:
@kwin during the repo initialisation process this method participates, IIRC the `SlingRepository` will not be available as service yet? (see https://github.com/Netcentric/accesscontroltool/blob/b8f5453ce9fed0ac953466d05d08da0c790a7248/accesscontroltool-bundle/src/main/java/biz/netcentric/cq/tools/actool/impl/AcInstallationServiceImpl.java#L103)

Reply:
right, this probably requires some refactoring of `AcInstallationServiceImpl()`